c10/core/SymInt.h

```diff
@@ -556,3 +556,26 @@ inline SymBool sym_ge(const SymInt& a, const SymInt& b) {
 }
 
 } // namespace c10
+
+#include <limits>
+
+namespace std {
+
+template <>
+class numeric_limits<c10::SymInt> {
+ public:
+  static constexpr bool is_specialized = true;
+
+  static constexpr int64_t max() noexcept {
+    return std::numeric_limits<int64_t>::max();
+  }
+
+  static constexpr int64_t min() noexcept {
+    return std::numeric_limits<int64_t>::min();
+  }
+
+  static constexpr bool is_signed = true;
+  static constexpr bool is_integer = true;
+};
+
+} // namespace std
```

Comment:
cc @ezyang to verify there's no unintended downsides to specializing `numeric_limits` for `c10::SymInt`? if this is too risky, I'll suggest some private templatized helper for getting the max in the conv check logic instead of using `numeric_limits` itself

Reply:
@thenumberouscode FYI the PyTorch Conference is going on at the moment so there may be some delay :)